src/gui/raster/qgsrasterattributetablewidget.cpp

```diff
@@ -0,0 +1,615 @@
+/***************************************************************************
+  qgsrasterattributetablewidget.cpp - QgsRasterAttributeTableWidget
+
+ ---------------------
+ begin                : 6.10.2022
+ copyright            : (C) 2022 by Alessandro Pasotti
+ email                : elpaso at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgsrasterattributetablewidget.h"
+#include "qgsrasterattributetable.h"
+#include "qgsrasterlayer.h"
+#include "qgsapplication.h"
+#include "qgsmessagebar.h"
+#include "qgsrasterattributetableaddcolumndialog.h"
+#include "qgsrasterattributetableaddrowdialog.h"
+#include "qgscolorbutton.h"
+#include "qgsgradientcolorrampdialog.h"
+#include "qgspalettedrasterrenderer.h"
+#include "qgssinglebandpseudocolorrenderer.h"
+#include "qgsrastershader.h"
+
+#include <QToolBar>
+#include <QAction>
+#include <QSortFilterProxyModel>
+#include <QMessageBox>
+#include <QFileDialog>
+
+QgsRasterAttributeTableWidget::QgsRasterAttributeTableWidget( QWidget *parent, QgsRasterLayer *rasterLayer, const int bandNumber )
+  : QgsPanelWidget( parent )
+  , mRasterLayer( rasterLayer )
+{
+  setupUi( this );
+
+  // Create the toolbar
+  QToolBar *editToolBar = new QToolBar( this );
+  editToolBar->setIconSize( QgsGuiUtils::iconSize() );
```

Comment:
```suggestion
  editToolBar->setIconSize( QgsGuiUtils::panelIconSize() );
```

Reply:
That was my first attempt, but `QSize GUI_EXPORT panelIconSize( QSize size );` requires a `size` and I don't know where I should get it from.